tests/rptest/services/admin_ops_fuzzer.py

```diff
@@ -495,6 +501,95 @@ def describe(self):
         }
 
 
+class DeleteRecords(Operation):
+    def __init__(self, prefix):
+        self.prefix = prefix
+        self.truncate_points = {}
+
+    def _random_truncate_points(self, ctx) -> Dict[str, Dict[int, int]]:
+        """
+        Returns a mapping of topics to partition offsets:
+        { "foo" : { 0 , 150 } } , will truncate foo partition 0 at offset 150
+        """
+        def deletable_topic_partitions(ctx):
+            def is_partition_empty(p):
+                return int(p.high_watermark) == 0 or (int(p.start_offset)
+                                                      == int(p.high_watermark))
+
+            def is_deletable(ctx, t):
+                cfgs = ctx.rpk().describe_topic_configs(topic=t)
+                (values, _) = cfgs['cleanup.policy']
+                values = values.split(',')
+                return 'delete' in values
+
+            deletable_topics = [
+                t for t in ctx.rpk().list_topics()
+                if is_deletable(ctx, t) and t.startswith(self.prefix)
+            ]
+            tps_with_data = {
+                t: [
+                    p for p in ctx.rpk().describe_topic(t)
+                    if not is_partition_empty(p)
+                ]
+                for t in deletable_topics
+            }
+            return {t: ps for t, ps in tps_with_data.items() if len(ps) > 0}
+
+        topic_partitions = deletable_topic_partitions(ctx)
+
+        def random_truncate_offset(p):
+            return random.randint(
+                int(p.start_offset) + 1, int(p.high_watermark))
```

Comment:
nit: seems like we also accept -1 to indicate truncating the entire partition. Might be worth including that at some probability.

Reply:
although that is true KCL does not accept that, when we move to using `--to-hwm` with the new `rpk trim` command that is how it will be passed